primary/src/header_waiter.rs

```diff
@@ -329,39 +326,42 @@ impl<PublicKey: VerifyingKey> HeaderWaiter<PublicKey> {
                         },
                         ReconfigureNotification::Shutdown => return
                     }
-                }
-            }
+                },
+
+                // Check for a new consensus round number
+                Ok(()) = self.rx_new_consensus_round.changed() => {
```

Comment:
I believe we might still need to have this outside of the `select` statement so GC doesn't run only when the round changes as requests as this will make GC run only once per round - while requests for earlier rounds might still come in.

That being said what we can probably do is keep the way you refactored it, and maybe immediately deliver any `header` either when parents or batches is requested in case the header's round is passed gc round logic so we don't necessarily need to wait until some GC run.

Reply:
Hey @akichidis -- I have moved it outside protected by a flag, but besides when the commit round advances, I am not sure under what other condition to trigger the GC. Any ideas?